### ISSUE TITLE
When the Setup from the Screens Change without Reboot (One More or less or a different Resolution) then it does not work on all Screens

### ISSUE THREAD

**Issue by RalfLuchsinger** [flameshot-org/flameshot#3597]
### Flameshot Version

Flameshot v12.1.0 (96c2c82e)

### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...)

### Operating System type and version

Windows 11 2303

### Description

We have a lot changes in the Setup with the Screens. The Day start with the Screen from the Surface alone (so the OS Boot with a single Screen), then you go to the Homeoffice with 1 External Screen, later to the Office with 2 External Screens and in the Meeting room you make miracast to another Screen. Everytime you change the Setup from the Screens Flameshot needs a restart to work properly - thats realy sad for the User Experience of this cool product!!!

### Steps to reproduce

run Flameshot at your Device with no Connection to another Screen. 
Flameshot works
Than Change to ad Docking Station with 2 External Screens
Flameshot does not work on the External Screens

### Screenshots or screen recordings

The Case is clear without Screenshots.

### System Information

The Case is clear without Screenshots.

**Comment by madrats** [comment]
Can confirm on Windows 10 ver. 22H2 with Flameshot v12.1.0

**Comment by MatanRubin** [comment]
Can confirm. Everytime I connect or disconnect from my external monitor I have to restart Flameshot to get it to work properly. 

Flameshot v12.1.0 (96c2c82e)
Compiled with Qt 5.15.5
darwin: 23.4.0
osx: 14.4

**Comment by madrats** [comment]
This issue is duplicated over several tickets:

- #3505 
- #3320 
- #3089 - according to this, bug traced to `ScreenGrabber::desktopGeometry()`
- #2610 
- #1506 

Hope this helps solve the problem quicker, until then just restart Flameshot

**Comment by borgmanJeremy** [comment]
@madrats  thank you so much for documenting all the duplicates. I was just going to do the same and this will save me a bunch of time. I plan to read through them later, pick the best "surrogate issue", and close duplicates.

**Comment by ASIJake** [comment]
Yes! Activity on this project! I hope we see a release sometime soon - it's been too long! 🙂

**Comment by mmahmoudian** [comment]
@borgmanJeremy can the monitor polling happens when the `gui` is triggered instead of when the daemon is started?

**Comment by borgmanJeremy** [comment]
@mmahmoudian That could work too. The Qt signal for the monitor being changed fires correctly. The issue is there are some variables in flameshot that are not properly reset.

**Comment by Ferlorin** [comment]
I noticed that on windows this happens when the main display is not the Left one. For example 
This works:
<img width="774" height="521" alt="Image" src="https://github.com/user-attachments/assets/2205a93c-20d3-4e78-af56-b3ebe1417a80" />

But this doesnt
<img width="736" height="528" alt="Image" src="https://github.com/user-attachments/assets/39c3c927-00da-4188-ae3c-679c7b2e2b8b" />

Hope this helps

**Comment by hugorn** [comment]
Same here on Cachyos Linux with Xorg
This happens when the main display is not the Left one.

**Comment by salamon** [comment]
Amazing tool! I installed on Win 11 and loved it, till i connected a second (left) monitor :(

**Comment by jerome-dl** [comment]
Hi team,

If an automatic detection of screen-configuration changes is not possible/easy, at least you could add a button for "refreshing configuration" to avoid a manual stop & restart operation (which is the current _poor_ workaround).

Thanks.

**Comment by cloudhal** [comment]
Yes we need a refresh button please, restarting every time monitors change is pretty annoying

**Comment by pipi-bear** [comment]
Same problem here on arch linux with wayland, two monitors (main display on the left).
The second display just couldn't be detected, I tried the restart approach but does not help.

I'm using:

Flameshot v13.1.0 ()
Compiled with Qt 6.9.1

Thanks!

**Comment by cloudhal** [comment]
I'm actually giving up on flameshot, I like everything else but I frequently change screen setups by docking my laptop or using remote desktop, and sometimes even restarting the app it is stuck in the wrong resolution and only captures half of the screen. Shame as everything else is great.

**Comment by TheCraftedGamer** [comment]
when using main display (right) and secondary display(left) 13.3.0 shows main as blank.
theres a temp fix to this, swap the main display. but, thats not much of a solution.
i reverted to 13.1.0 and everything works now.
i did not test 13.2.0.

dunno if this was posted but i came from a *duplicate* post. 
Flameshot v13.1.0 (79395fd3)
Compiled with Qt 6.8.1
winnt: 10.0.26100
windows: 11

**Comment by Nashevil** [comment]
Same problem with dual monitor, I can't take a screenshot, when I try to take a screenshot on the third screen, the image from the first screen (the laptop screen) shifts, forcing me to move what I want to capture to the laptop screen to get a smaller screenshot, and then having to take the screenshot twice.

I've tried restarting Flameshot several times, both before and after connecting the dock station, and it does the same thing. This didn't happen with version 12.

<img width="4319" height="803" alt="Image" src="https://github.com/user-attachments/assets/8e70178f-7f4f-4202-85bc-6d9fd060e483" />

Flameshot v13.1.0
windows: 11 Pro 25H2

**Comment by meichthys** [comment]
I'm also seeing this on ARCH Linux (KDE). Secondary monitor works, primary monitor does not:

<img width="487" height="388" alt="Image" src="https://github.com/user-attachments/assets/2e96f424-7dfb-41ea-9e98-72c93a147390" />

**Comment by mmahmoudian** [comment]
@cloudhal  @TheCraftedGamer @Nashevil your issues are not related tobthis thread. Luckily, the issue you are facing is already addressed in #4498 . I will hide your comment as off-topic.
